apps/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,25 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgTemplateOutlet } from '@angular/common';
+import {
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
+import { Item } from '../../model/item.model';
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  styleUrls: ['./card.component.scss'],
+  imports: [NgFor, ListItemComponent, NgTemplateOutlet],
 })
 export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
+  @Input() items: Item[] | null = [];
   @Input() customClass = '';
-
-  CardType = CardType;
-
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore
-  ) {}
-
-  addNewItem() {
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
-  }
+  @Input() listItemTemplate!: TemplateRef<any>;
```

Comment:
getting closer. 

You can user @ContentChild to get your template ref from your parent. 

and you should use listItemComponent in the parent : 
- you will not have an output going thought each component
- you can add ng-content inside this component and set whatever you want inside instead of concatening string inside name property.